### DISCUSSION TITLE
DMA interrupts?

### DISCUSSION THREAD

**Discussion by sciencedude1990** [micropython/micropython#10694] [RP2040 / Pico]
Hi there,

Great work on the Micropython port for the RP2040 / Pico!

I am working with the DMA, but can't figure out if Micropython supports the interrupts that the DMA generates.

Does Micropython support DMA interrupts?

Thanks!

**Comment by rkompass** [comment]
I raised this question within a show and tell [discussion contribution](https://github.com/micropython/micropython/discussions/9169), which might be helpful for you (?.?), but got no answer. 

I suppose the answer is no. Perhaps we should raise an issue as a feature request.

**Reply by sciencedude1990** [reply]
I raised "DMA interrupts #10696" as a feature request.  I also dropped some money in the contribution box - not sure if those two things influence each other...  :)

**Reply by robert-hh** [reply]
So there is @markb139, who tries to put a PR #10695 for DMA support online. That one supports DMA with an option for calling a callback when the DMA is finished. @sciencedude1990 , is that something you mean with DMA interrupts?

**Reply by sciencedude1990** [reply]
I jumped in and tried to review the code.  It looks like the interrupts are hard coded to use only DMA interrupt 0 (I think there are two interrupts available, 0 and 1).

**Reply by markb139** [reply]
Yes, I’ll update the code. I wrote this code last December to help play audio via the PWM so I could make a Christmas decoration play tunes:)

**Comment by samskiter** [comment]
Looks like this is still outstanding?
